### ISSUE TITLE
SysV shared memory segment leaks on nginx reload

### ISSUE THREAD

**Issue by hiboma** [matsumoto-r/mruby-mutex#3]
Hi. 

I found shared memory segment leaks on nginx reload.
# How to reproduce?
## nginx.conf

```
events {
  worker_connections  128;
}

daemon off;
error_log  /dev/stderr debug;

http {
    mruby_init_code '
      mutex = Mutex.new :global => true
    ';

    server {
        listen       8080;
        server_name  localhost;
    }
}
```
## check ipcs -m

After nginx started, check `ipcs -m`. 

```
$ ipcs -m
```

```
$ LANG=C ipcs -m

------ Shared Memory Segments --------
key        shmid      owner      perms      bytes      nattch     status      
0x00000000 622592     vagrant    600        40         2                       
```
## reload nginx

```
kill -HUP $( pgrep -f 'nginx: master' )
```
## check ipcs -m, again

```
$ LANG=C ipcs -m

------ Shared Memory Segments --------
key        shmid      owner      perms      bytes      nattch     status      
0x00000000 622592     vagrant    600        40         2                       
0x00000000 655361     vagrant    600        40         2                       
```
## Environemnt

I have tested on CentOS7 

```
$ uname -a
Linux lb.minne.pbdev 3.10.0-123.20.1.el7.x86_64 #1 SMP Thu Jan 29 18:05:33 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/redhat-release 
CentOS Linux release 7.0.1406 (Core) 
```

---

I'll send a PR, please wait :tea:

**Comment by matsumotory** [comment]
Hi!! Thank you for your report. 

> please wait

Yes!

**Comment by matsumotory** [comment]
Fixed via #4
